tls/data_source_public_key_test.go

```diff
@@ -0,0 +1,64 @@
+package tls
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+const (
+	expectedPublic = `-----BEGIN PUBLIC KEY-----
+MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQDPLaq43D9C596ko9yQipWUf2Fb
+RhFs18D3wBDBqXLIoP7W3rm5S292/JiNPa+mX76IYFF416zTBGG9J5w4d4VFrROn
+8IuMWqHgdXsCUf2szN7EnJcVBsBzTxxWqz4DjX315vbm/PFOLlKzC0Ngs4h1iDiC
+D9Hk2MajZuFnJiqj1QIDAQAB
+-----END PUBLIC KEY-----`
+	expectedPublicSSH = `ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAAAgQDPLaq43D9C596ko9yQipWUf2FbRhFs18D3wBDBqXLIoP7W3rm5S292/JiNPa+mX76IYFF416zTBGG9J5w4d4VFrROn8IuMWqHgdXsCUf2szN7EnJcVBsBzTxxWqz4DjX315vbm/PFOLlKzC0Ngs4h1iDiCD9Hk2MajZuFnJiqj1Q==`
+)
+
+func TestAccPublicKey_dataSource(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		Providers: testProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: fmt.Sprintf(testAccDataSourcePublicKeyConfig, testPrivateKey),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("data.tls_public_key.test", "public_key_pem", strings.TrimSpace(expectedPublic)+"\n"),
+					resource.TestCheckResourceAttr("data.tls_public_key.test", "public_key_openssh", strings.TrimSpace(expectedPublicSSH)+"\n"),
+				),
+			},
+			resource.TestStep{
+				Config: `
+					resource "tls_private_key" "test" {
+						algorithm = "RSA"
+					}
+					output "private_key_pem" {
```

Comment:
Are these outputs actually being tested anywhere? I don't see any checks for them...

Can this test be broken out into another test? As it's pretty much a completely different test case (manually supplied key versus generated one).

Reply:
The output can be removed indeed. Since the test was for all Public Key, I thought they belonged within the same function, but as separate steps. Is that not the right approach?